buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java

```diff
@@ -34,8 +34,9 @@
 
 public class Jdk implements Buildable, Iterable<File> {
 
-    static final Pattern VERSION_PATTERN = Pattern.compile("(\\d+)(\\.\\d+\\.\\d+)?\\+(\\d+)(@([a-f0-9]{32}))?");
-    private static final List<String> ALLOWED_PLATFORMS = Collections.unmodifiableList(Arrays.asList("linux", "windows", "darwin"));
+    static final Pattern VERSION_PATTERN =
+        Pattern.compile("(adoptopenjdk|openjdk)-(\\d+)(\\.\\d+\\.\\d+)?\\+(\\d+(?:\\.\\d+)?)(@([a-f0-9]{32}))?");
```

Comment:
Instead of overloading the version with even more information, and we make this a separate property of Jdk? I think this fits better alongside platform and version rather than an artificial part of the latter.

Reply:
👍We should model this as a first-class thing in `Jdk`. FWIW, I'd like to avoid having anymore things called "distribution" in our build logic, can we call this axis of the JDK "vendor" or something similar?

Reply:
I pushed 4964fb6fa7973c38e641a97780e9b2ff72e1e0a0.